exercises/concept/lasagna-master/.meta/exemplar.js

```diff
@@ -0,0 +1,89 @@
+/// <reference path="./global.d.ts" />
+// @ts-check
+
+/**
+ * Determines whether the lasagna is done based on the
+ * remaining time on the timer.
+ *
+ * @param {number} time time left on the timer
+ * @returns {string} cooking status
+ */
+export function cookingStatus(time) {
+  if (time === undefined) {
+    return 'You forgot to set the timer.';
+  }
+
+  if (time === 0) {
+    return 'Lasagna is done.';
+  }
+
+  return 'Not done, please wait.';
+}
+
+/**
+ * Estimates the preparation time based on the number of layers.
+ *
+ * @param {string[]} layers
+ * @param {number} avgPrepTime
+ * @returns {number} total preparation time
+ */
+export function preparationTime(layers, avgPrepTime = 2) {
+  return layers.length * avgPrepTime;
+}
+
+/**
+ * Calculates how many noodles and much sauce are needed for the
+ * given layers.
+ *
+ * @param {string[]} layers
+ * @returns {Quantities} quantities needed for the given layers
+ */
+export function quantities(layers) {
+  let noodles = 0;
+  let sauce = 0;
+
+  for (let i = 0; i < layers.length; i++) {
+    if (layers[i] === 'noodles') {
+      noodles += 50;
+    }
+
+    if (layers[i] === 'sauce') {
+      sauce += 0.2;
+    }
+  }
+
+  return { noodles, sauce };
+}
+
+/**
+ * Adds the secret ingredient from the ingredient list that a
+ * friend provided to your ingredient list.
+ *
+ * @param {string[]} friendsList
+ * @param {string[]} myList
+ */
+export function addSecretIngredient(friendsList, myList) {
+  const lastIndex = friendsList.length - 1;
+  myList.push(friendsList[lastIndex]);
+}
+
+/**
+ * Calculates the amounts of ingredients needed for a certain
+ * amount of portions.
+ * Assumes the original amounts were meant for 2 portions.
+ * Does not modify the original recipe.
+ *
+ * @param {Entry<string, number>} recipe
```

Comment:
My IDE says `Unresolved variable or type 'Entry'` but I don't know enough about JSDoc to say why or what would be the correct type. I guess in TS in would be `Record`?

Reply:
Yes, it should be `Record`

Reply:
:see_no_evil: I thought any name is fine there.

Reply:
It is, but it just doesn't mean anything in _any_ case :p